docs/shared/tutorials/npm-workspaces.md

```diff
@@ -0,0 +1,377 @@
+
+```json {% fileName="nx.json" %}
+{
+  "targetDefaults": {
+    "build": {
+      "dependsOn": ["^build"]
+    }
+  }
+}
+```
+
+This configuration means that if you run `build` on any project, Nx will first run `build` for the dependencies of that project and then run `build` on the project itself. The `^build` text means "the `build` tasks of the project's dependencies." You can visualize this in the Nx graph by selecting the `Tasks` dropdown in the top left and clicking `Show all tasks`:
+
+```shell {% path="~/tuskydesigns" %}
+npx nx graph
+```
+
+{% graph height="200px" title="Build Task Pipeline" type="task" jsonFile="shared/tutorials/npm-workspaces-build-tasks1.json" %}
+{% /graph %}
+
+With this pipeline in place, you will never run into the error we hit at the beginning of the tutorial where the `forms` and `buttons` packages weren't built so the `demo` app couldn't build. Test this out by deleting the `packages/forms/dist` folder and then re-running the `build` task for the `demo` app.
+
+```text {% command="npx nx build @tuskdesign/demo" path="~/tuskydesigns" %}
+...
+
+ NX   Successfully ran target build for project @tuskdesign/demo and 2 tasks it depends on (40ms)
+
+Nx read the output from the cache instead of running the command for 3 out of 3 tasks.
+```
+
+Not only does the build complete successfully, but it finishes instantly and the `packages/forms/dist` folder is put back in place thanks to the caching.
+
+## Create a Task Pipeline
+
+You may have noticed in the `apps/demo/package.json` file, there is a `prebuild` script that runs `typecheck` before the `build` script in order to catch any type errors. Let's set up this same behavior in the Nx task pipeline as well.
+
+```json {% fileName="nx.json" highlightLines=[5] %}
+{
+  "$schema": "./node_modules/nx/schemas/nx-schema.json",
+  "targetDefaults": {
+    "build": {
+      "dependsOn": ["^build", "typecheck"],
+      "outputs": ["{projectRoot}/dist"],
+      "cache": true
+    },
+    "typecheck": {
+      "cache": true
+    },
+    "lint": {
+      "cache": true
+    }
+  },
+  "defaultBase": "main"
+}
+```
+
+The `dependsOn` line makes Nx run the `typecheck` task for the current project and the `build` task for any dependencies before running the current project's `build` task. Now `nx build` will run the `typecheck` task just like `npm run build` does.
+
+## Use Nx Plugins to Enhance Vite Tasks with Caching
+
+You may remember that we defined the `outputs` property in `nx.json` when we were answering questions in the `nx init` script. The value is currently hard-coded so that if you change the output path in your `vite.config.ts`, you have to remember to also change the `outputs` array in the `build` task configuration. This is where plugins can help.
```

Comment:
I missed this the first round of review, but I feel like we need to emphasize the value proposition of plugins much more here. Like that last sentence "This is where plugins can help"..but transform it into something more like:

"This is where Nx plugins help. They directly infer information from the actual tooling configs like `vite.config.ts` in this case. Nx plugins...

- automatically configure caching for you, including inputs and outputs based on the underlying tooling config
- infer tasks ....."

something along those lines. wdyt?